### ISSUE TITLE
[Bug] Black screen on startup

### ISSUE THREAD

**Issue by halotroop2288** [blawar/goldbricks#7]
# Black screen on startup

## Bug description

Something has changed within either Atmosphére or HOS which makes GoldBricks launch into an unrecoverable black screen. Sometimes this causes the entire system to freeze indefinitely.

## Your setup

- **Console firmware**: 11.0.1

- **CFW with version**: Atmosphére 0.17.0

- **Goldbricks version**: 1.0.0

## Bug reproduction

1. Update Switch to firmware version `11.0.1`.
2. Update Atmosphére to version `0.17.0`.
3. Install Goldbricks version `1.0.0`.
4. Attempt to launch Goldbricks via title injection.
5. Observe a blank black screen which cannot be recovered except (sometimes) by pressing the home button and closing the injected title.

# Additional Information
- Sometimes this causes the system to freeze, in which case the only way to recover is to power off the system using a forced shutdown and restart it.
  - This usually only happens on the second try before rebooting the system.

**Comment by gaoruhao** [comment]
I ran into the same issue. The only difference is atmosphere version is 0.17.1.

**Comment by halotroop2288** [comment]
> atmosphere version is 0.17.1

Yeah. I'm sure the problem will persist for future versions, so it's something that will hopefully be looked into. I'm not so sure though, seeing as this app hasn't been updated since 2019.

**Comment by RQWorldblender** [comment]
The upstream Goldleaf has released a new version which will work on HOS 11.0.0 and later, since February 8: https://github.com/XorTroll/Goldleaf/releases/tag/0.9
It's that only installer that I got to successfully run so far, running on Atmosphere 0.18.1 with Mesosphere. I wish I could start my own fork, so that I can integrate at least NSZ support, but I may be doing excessive work in case Goldbricks gets worked on again.

**Comment by blawar** [comment]
It will be updated soon.

**Comment by RQWorldblender** [comment]
> It will be updated soon.

Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.

**Comment by blawar** [comment]
> > It will be updated soon.
> 
> Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
> It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.

If you submit a PR here, it will be accepted.  Otherwise I will update it.

**Comment by RQWorldblender** [comment]
> It will be updated soon.

Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.

> > It will be updated soon.
> 
> Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
> It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.

It's hard for me to make a PR here, because I forked Goldleaf instead of Goldbricks, as all the changes have been made over there.

**Comment by blawar** [comment]
> > It will be updated soon.
> 
> Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
> It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.
> 
> > > It will be updated soon.
> > 
> > 
> > Well, I did that faster than you could with my own fork of Goldleaf 0.9: https://github.com/RQWorldblender/Goldleaf
> > It mainly adds your NSZ support code, making changes as needed. I have not yet tested the patched code as of this writing.
> 
> It's hard for me to make a PR here, because I forked Goldleaf instead of Goldbricks, as all the changes have been made over there.

Goldbricks was forked from goldleaf at 651c1ed4e0611cd, assuming  goldleaf did not change its git history, you would just pull/merge from upstream then resolve the conflicts.

**Comment by RQWorldblender** [comment]
> Goldbricks was forked from goldleaf at [651c1ed](https://github.com/blawar/goldbricks/commit/651c1ed4e0611cdb3284a7c7c3e72f425f64110e), assuming goldleaf did not change its git history, you would just pull/merge from upstream then resolve the conflicts.

There is a lot of code cleanup that happened since you forked Goldleaf that would make it hard for me to do a simple merge/pull. It took me a while to find the appropriate place to insert your code. I merely took action because you didn't update Goldbricks at all last year.

**Comment by blawar** [comment]
> > Goldbricks was forked from goldleaf at [651c1ed](https://github.com/blawar/goldbricks/commit/651c1ed4e0611cdb3284a7c7c3e72f425f64110e), assuming goldleaf did not change its git history, you would just pull/merge from upstream then resolve the conflicts.
> 
> There is a lot of code cleanup that happened since you forked Goldleaf that would make it hard for me to do a simple merge/pull. It took me a while to find the appropriate place to insert your code. I merely took action because you didn't update Goldbricks at all last year.

Goldbricks was not updated last year because Goldleaf was not updated last year.  0.9 was only just recently released, before that it was a private beta build for the longest time.

Its no worries though, I'll update goldbricks soon.

**Comment by RQWorldblender** [comment]
> Goldbricks was not updated last year because Goldleaf was not updated last year. 0.9 was only just recently released, before that it was a private beta build for the longest time.
> 
> Its no worries though, I'll update goldbricks soon.

I hope you will soon, because I basically have few options for installing NSZ files on firmware 11.0.0 and later. Feel free to look at my fork for inspiration on what you can do.

Make sure to update the GBATemp thread once you do make an update.

**Comment by RQWorldblender** [comment]
@blawar I think I made it faster in my fork to patch in NSZ support than you could. Here is a build made from the most recent commit found in my fork.
[Goldleaf.nro.zip](https://github.com/blawar/goldbricks/files/6065892/Goldleaf.nro.zip)

**Comment by halotroop2288** [comment]
What's taking so long? Is it too complicated to merge the fix from Goldleaf into Goldbricks?

**Comment by waltercool** [comment]
Soon = Never lol

**Comment by JonJaded** [comment]
Damn.
